Library/Homebrew/formula.rb

```diff
@@ -1853,12 +1853,21 @@ def self.formulae_with_cask_dependents(casks)
          .flat_map { |f| [f, *f.runtime_formula_dependencies].compact }
   end
 
-  # An array of all installed {Formula} without {Formula} dependents
+  # An array of all installed {Formula} without runtime {Formula}
+  # dependents for bottles and without build {Formula} dependents
+  # for those built from source.
   # @private
   def self.formulae_with_no_formula_dependents(formulae)
     return [] if formulae.blank?
 
-    formulae - formulae.flat_map(&:runtime_formula_dependencies)
+    formulae - formulae.each_with_object([]) do |formula, dependents|
+      dependents.concat(formula.runtime_formula_dependencies)
```

Comment:
```suggestion
      dependents += formula.runtime_formula_dependencies
```

Reply:
Is this just a style thing? From what I understand `Array#concat` is in-place while `Array#+=` always creates a new array. It probably doesn't make a big difference either way but I'm just interested in why. Happy to change it though.